channels/layers.py

```diff
@@ -242,14 +243,16 @@ async def receive(self, channel):
         assert self.valid_channel_name(channel)
         self._clean_expired()
 
-        queue = self.channels.setdefault(channel, asyncio.Queue())
+        queue = self.channels.setdefault(
+            channel, asyncio.Queue(maxsize=self.get_capacity(channel))
+        )
 
         # Do a plain direct receive
         try:
             _, message = await queue.get()
         finally:
             if queue.empty():
-                del self.channels[channel]
+                self.channels.pop(channel, None)
```

Comment:
Can you explain why this is an improvement? I'm not sure this is an improvement as it slightly hurts code readability IMO

(applies to all replacements of `del` in this PR)

Reply:
it is safer. If queue empty provides the wrong result, the code doesn't fail.

This can happen in async scenarios

Reply:
Can you provide a regression test to make sure this doesn't happen again in the future?

Reply:
done